### PR TITLE
[YUNIKORN-356] Add e2e tests for state aware app scheduling

### PR DESCRIPTION
**Pull request by ayubpathan**

### CODE REVIEW

**Comment by ayubpathan** [comment]
@yangwwei  Please review these e2e test changes! CodeCov impact seems like not related.

File: test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	var c, err = k.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
+		configmanager.DefaultYuniKornConfigMap)
+	Ω(err).NotTo(HaveOccurred())
+	Ω(c).NotTo(BeNil())
+
+	oldConfigMap = c.DeepCopy()
+	Ω(c).Should(BeEquivalentTo(oldConfigMap))
+	var stateAwareStr = "partitions:\n  -\n    name: default\n    placementrules:\n      - name: tag\n        " +
+		"value: namespace\n        create: true\n    queues:\n      - name: root\n        " +
+		"submitacl: '*'\n        properties:\n          application.sort.policy: stateaware"
```
Comment (yangwwei) [inline test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go:56]: Can we use a more readable form for such multi-line strings?
Such as https://yourbasic.org/golang/multiline-string/#raw-string-literals

File: test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/apache/incubator-yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+
+	"github.com/onsi/ginkgo/reporters"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	"github.com/apache/incubator-yunikorn-k8shim/test/e2e/framework/configmanager"
```
[the remark this answers is not in the document]
Comment (ayubpathan) [reply test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go]: Done

File: test/e2e/state_aware_app_scheduling/fallback_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		By("Verify that the job is scheduled by YuniKorn & is in starting state")
+		Ω(appsInfo["applicationState"]).To(Equal("Starting"))
+		Ω("yunikorn").To(Equal(sleepRespPod.Spec.SchedulerName))
+	}, 60)
+
+	It("Verify_App_State_Transition_To_Running_Post_Timeout", func() {
+		By("Wait for fallback timeout of 5mins")
+		err = restClient.WaitForAppStateTransition(sleepRespPod.ObjectMeta.Labels["applicationId"],
+			yunikorn.States().Application.Running,
+			360)
```
[the remark this answers is not in the document]
Comment (ayubpathan) [reply test/e2e/state_aware_app_scheduling/fallback_test.go]: Increased it to 6 minutes..

File: test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go (hunk 1 above)
Comment (ayubpathan) [reply test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go]: Changed it!

**Comment by ayubpathan** [comment]
Thanks @yangwwei for the review.  I have addressed the comments, please review/merge.

**Comment by yangwwei** [comment]
LGTM, merged, thanks @ayubpathan

**Pull request merged**